lein-template/src/leiningen/new/formatting.clj

```diff
@@ -0,0 +1,28 @@
+(ns leiningen.new.formatting
+  (:require [clojure.pprint :refer [pprint]]
+            [clojure.string])
+  (:import [java.io StringWriter]))
+
+(defn- pprint-to-str [v]
+  (let [out (StringWriter.)]
+    (pprint v out)
+    (clojure.string/trim-newline (.toString out))))
+
+(defn parse-map [map]
+  (apply concat
+         (mapv
+           (fn [[k v]]
+             (cond
+               (empty? v) [(str k " " v)]
+               (string? v) [(str k v)]
+               :else [(str k)
+                      (pprint-to-str v)]))
+           map)))
+
+(defn parse-cascading-routes [routes]
```

Comment:
This function can probably be removed. I don't think we need to write special code around such specific behaviour.